httpgrpc/httpgrpc.go

```diff
@@ -63,6 +65,29 @@ func HTTPResponseFromError(err error) (*HTTPResponse, bool) {
 	return &resp, true
 }
 
+// statusFromError tries to cast the given error into status.Status.
+// If the given error, or any error from its tree are a status.Status,
+// that status.Status and the outcome true are returned.
+// Otherwise, nil and the outcome false are returned.
+// This implementation differs from status.FromError() because the
```

Comment:
It does check that. The implementation is nearly the same as this method:

https://github.com/grafana/mimir/blob/main/vendor/google.golang.org/grpc/status/status.go#L113

Reply:
@56quarters you linked the grpc’s status, and not gogo’s status.

Reply:
OK, but that's the one we should be using? That's the one that the circuit breakers use to extract gRPC error codes.

Reply:
Hi @56quarters, 
Curently gogo is used both in `dskit` and in `mimir`. It is used even for the generation of Go structs out of `.proto` sources (e.g., [here](https://github.com/grafana/dskit/blob/main/Makefile#L63-L67)).
I agree that we should replace usages of gogo's `status` package with grpc's `status` package, but this should be done in a separate PR, because that change would be very big.

For the moment I changed the implementation of the `statusFromError` and made it use `grpcstatus.FromError()`, since these 2 implementations are, indeed, similar.

WDYT?

Reply:
Update: apparently we already have an [issue](https://github.com/weaveworks/common/issues/268) for removing the deprecated `gogo` dependency. But this would be a breaking change, plus we don't know which package to use instead. The `grpc` package has some big performance issues, that's why the `gogo` project was created.

> That's the one that the circuit breakers in Mimir use to extract gRPC error codes
This is not a problem, since gRPC errors created by both `grpc` and `gogo` can be recognized by each other.